src/Cache.hs

```diff
@@ -0,0 +1,77 @@
+module Cache (createCache, cacheRead, cacheWrite, cacheInvalidate) where
+
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Definitions
+
+data Cache = Cache {
+    cacheSize :: Int, 
+    associativity :: Int, 
+    blockSize :: Int, 
+    cacheContent :: Map (Int, Int) CacheBlock
+} deriving (Show)
+
+data CacheBlock = CacheBlock {
+    isBlockValid :: Bool, 
+    isBlockDirty :: Bool, 
+    blockContent :: [Int]
+} deriving (Show)
+
+type MemoryAddress = Int
+type IsReadHit = Bool
+type IsWriteHit = Bool
+
+main = print (
+    cacheInvalidate (
```

Comment:
The usual haskell idiom for this would be
`print $ cacheInvalidate $ fst $ cacheRead (createCache 64 2 4) 0x00000001` 
The `$` operator has the lowest precedence and will always evaluate the right side before the left

Reply:
Actually i realize another idiomatic way could be:

`(print . cacheInvalidate . fst . cacheRead) (createCache 64 2 4) 0x00000001`